sonar-php-plugin/src/test/java/org/sonar/plugins/php/PHPSensorTest.java

```diff
@@ -587,6 +588,53 @@ public void should_use_test_file_checks() {
     assertThat(check.wasTriggered).isTrue();
   }
 
+  @Test
+  public void should_not_analyze_unchanged_file_if_setting_is_enabled() {
+    checkFactory = new CheckFactory(getActiveRules());
+    context.fileSystem().add(inputFile(ANALYZED_FILE, Type.MAIN, InputFile.Status.SAME));
+    context.setCanSkipUnchangedFiles(true);
+    createSensor().execute(context);
+    assertThat(context.allIssues()).isEmpty();
+  }
+
+  @Test
+  public void should_analyze_file_without_status_if_setting_is_enabled() {
+    checkFactory = new CheckFactory(getActiveRules());
+    context.fileSystem().add(inputFile(ANALYZED_FILE, Type.MAIN, null));
+    context.setCanSkipUnchangedFiles(true);
+    createSensor().execute(context);
+    assertThat(context.allIssues()).isNotEmpty();
+  }
+
+  @Test
+  public void should_analyze_changed_file_if_setting_is_enabled() {
+    checkFactory = new CheckFactory(getActiveRules());
+    context.fileSystem().add(inputFile(ANALYZED_FILE, Type.MAIN, InputFile.Status.CHANGED));
+    context.setCanSkipUnchangedFiles(true);
+    createSensor().execute(context);
+    assertThat(context.allIssues()).isNotEmpty();
+  }
+
+  @Test
+  public void should_not_analyze_unchanged_file_if_internal_property_is_set() {
+    checkFactory = new CheckFactory(getActiveRules());
+    context.fileSystem().add(inputFile(ANALYZED_FILE, Type.MAIN, InputFile.Status.SAME));
+    context.setSettings(new MapSettings().setProperty("sonar.php.skipUnchanged", "true"));
+    createSensor().execute(context);
+    assertThat(context.allIssues()).isEmpty();
+  }
+
+  @Test
+  public void should_analyze_unchanged_file_in_sonarlint_context() {
+    checkFactory = new CheckFactory(getActiveRules());
+    context.fileSystem().add(inputFile(ANALYZED_FILE, Type.MAIN, InputFile.Status.SAME));
+    context.setSettings(new MapSettings().setProperty("sonar.php.skipUnchanged", "true"));
+    context.setRuntime(SONARLINT_RUNTIME);
+    createSensor().execute(context);
+    assertThat(context.allIssues()).isNotEmpty();
+  }
+
```

Comment:
I would suggest adding a test where the flag `sonar.php.skipUnchanged` is explicitly set to `false`.